sunbird-es-utils/src/main/java/org/sunbird/common/ElasticSearchRestHighImpl.java

```diff
@@ -0,0 +1,907 @@
+            FieldSortBuilder mySort =
+                new FieldSortBuilder(entry.getKey() + ElasticSearchHelper.RAW_APPEND)
+                    .setNestedFilter(
+                        new TermQueryBuilder(dateMapEntry.getKey(), dateMapEntry.getValue()))
+                    .sortMode(SortMode.MIN)
+                    .order(ElasticSearchHelper.getSortOrder((String) map.get(JsonKey.ORDER)));
+            searchSourceBuilder.sort(mySort);
+          }
+        }
+      }
+    }
+
+    // apply the fields filter
+    searchSourceBuilder.fetchSource(
+        searchDTO.getFields() != null
+            ? searchDTO.getFields().stream().toArray(String[]::new)
+            : null,
+        searchDTO.getExcludedFields() != null
+            ? searchDTO.getExcludedFields().stream().toArray(String[]::new)
+            : null);
+
+    // setting the offset
+    if (searchDTO.getOffset() != null) {
+      searchSourceBuilder.from(searchDTO.getOffset());
+    }
+
+    // setting the limit
+    if (searchDTO.getLimit() != null) {
+      searchSourceBuilder.size(searchDTO.getLimit());
+    }
+    // apply additional properties
+    if (searchDTO.getAdditionalProperties() != null
+        && searchDTO.getAdditionalProperties().size() > 0) {
+      for (Map.Entry<String, Object> entry : searchDTO.getAdditionalProperties().entrySet()) {
+        ElasticSearchHelper.addAdditionalProperties(query, entry, constraintsMap);
+      }
+    }
+
+    // set final query to search request builder
+    searchSourceBuilder.query(query);
+    List finalFacetList = new ArrayList();
+
+    if (null != searchDTO.getFacets() && !searchDTO.getFacets().isEmpty()) {
+      searchSourceBuilder = addAggregations(searchSourceBuilder, searchDTO.getFacets());
+    }
+    ProjectLogger.log(
+        "ElasticSearchRestHighImpl:search calling search builder======"
+            + searchSourceBuilder.toString(),
+        LoggerEnum.INFO.name());
+
+    searchRequest.source(searchSourceBuilder);
+    Promise<Map<String, Object>> promise = Futures.promise();
+
+    ActionListener<SearchResponse> listener =
+        new ActionListener<SearchResponse>() {
+          @Override
+          public void onResponse(SearchResponse response) {
+            if (response.getHits() == null || response.getHits().getTotalHits() == 0) {
+              promise.success(new HashMap<>());
+            } else {
+              List<Map<String, Object>> esSource = new ArrayList<>();
+              Map<String, Object> responseMap = new HashMap<>();
+              long count = 0;
+              if (response != null) {
+                SearchHits hits = response.getHits();
+                count = hits.getTotalHits();
+
+                for (SearchHit hit : hits) {
+                  esSource.add(hit.getSourceAsMap());
+                }
+
+                // fetch aggregations aggregations
+                if (null != searchDTO.getFacets() && !searchDTO.getFacets().isEmpty()) {
```

Comment:
Complete if condition logic can be moved to some common place , from where both impl can use it. Hopping both Impl is using same logic

Reply:
done